### PR TITLE
Integrating travis-CI

### LINKED ISSUE #8: Connect it with Travis CI
Use travis to test the project automatically.

### PR DESCRIPTION
**Pull request by JOHNKYON**
Resolve #8.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+language: python
+python:
+  - "3.6"
+
+install:
+  - pip install -r requirements.txt
+
+script:
+  - pytest
```
Comment (ZhuGongpu) [inline .travis.yml]: Ensure an empty line at EOF.

Should also enforce an `.editorconfig` file.

File: test/test_null.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+"""This is the test case that do nothing, it's just used to pass travis-CI when no tests are
+available"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+
+def testnull():
+    """Test case that do nothing"""
```
Comment (ZhuGongpu) [inline test/test_null.py:10]: `Keep pytest happy`.

And add a issue ref on top.

**Review by ZhuGongpu — CHANGES_REQUESTED** [review]

File: test/test_null.py (hunk 2 above)
Comment (JOHNKYON) [reply test/test_null.py:10]: Refuse issue ref.
Looks silly.

File: .travis.yml (hunk 1 above)
Comment (JOHNKYON) [reply .travis.yml]: Leave .editconfig to another issue

Comment (JOHNKYON) [reply .travis.yml]: Done with EOF.

File: .gitignore [hunk 3]
```diff
@@ -130,6 +130,7 @@ pip-selfcheck.json
 ### JetBrains template
 # Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and WebStorm
 # Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+.idea
 
 # User-specific stuff
 .idea/**/workspace.xml
```
Comment (ZhuGongpu) [inline .gitignore]: Redundant.

File: test/test_null.py (hunk 2 above)
Comment (ZhuGongpu) [reply test/test_null.py:10]: Adding references is designated for **readability** and **maintenance**.

**Review by ZhuGongpu — CHANGES_REQUESTED** [review]

File: test/test_null.py (hunk 2 above)
Comment (JOHNKYON) [reply test/test_null.py:10]: Outer links will only decrease the readability. What if one day github gets **blocked** by the firewall?

File: .gitignore [hunk 4]
```diff
@@ -132,51 +132,3 @@ pip-selfcheck.json
 # Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
 .idea
 
-# User-specific stuff
```
Comment (ZhuGongpu) [inline .gitignore]: Should do this in the last commit.

File: test/test_null.py (hunk 2 above)
Comment (ZhuGongpu) [reply test/test_null.py:10]: That is not the point.

Outer links, which may be GitHub Issues, StackOverflow or anything else, are just references telling some basic background information and design principles.

**Review by ZhuGongpu — CHANGES_REQUESTED** [review]

File: test/test_null.py (hunk 2 above)
Comment (JOHNKYON) [reply test/test_null.py:10]: Which would decrease the robust. Robust code should be maintained by **non-specific** environment while too much information that does not stick to the code itself would distract people. Use GitHub Issues or anything like that to help develop, but never put everything on it. The code should be  **complete** even when all developing tools are removed.

**Pull request merged**